File: spring/src/main/java/hu/marko/szakdolgozat/spring/service/NewsfeedService.java
package hu.marko.szakdolgozat.spring.service;

import hu.marko.szakdolgozat.spring.service.model.Newsfeed;
import hu.marko.szakdolgozat.spring.service.model.PageModel;

public interface NewsfeedService {
  PageModel<Newsfeed> findByPageAndSizeAndFilterAndOrder(Integer page, Integer size, String filter, String order,
      Boolean ascendingDirection);

  PageModel<Newsfeed> findByUserAndPageAndSizeAndFilterAndOrder(Long userid, Integer page, Integer size, String filter,
      String order, Boolean ascendingDirection);

  Newsfeed findOne(Long id);

  Newsfeed save(Newsfeed newsfeed);

  Boolean update(Long id, Newsfeed newsfeed);

  Long remove(Long id);
}

File: fastify/src/service/NewsFeedService.ts
import { Newsfeed } from '../entity/Newsfeed';

export interface iNewsFeedService {
  findByPageAndSizeAndFilterAndOrder(page: number, size: number, filter?: string, order?: string, ascendingDirection?: boolean): Promise<[Newsfeed[], number]>;
  findByUserAndPageAndSizeAndFilterAndOrder(userId: number, page: number, size: number, filter?: string, order?: string, ascendingDirection?: boolean): Promise<[Newsfeed[], number]>
  findOne(id: number): Promise<Newsfeed>;
  save(entity: Newsfeed): Promise<Newsfeed>;
  update(id: number, entity: Newsfeed): Promise<boolean>;
  remove(id: number): Promise<number>;
}